sympy/ntheory/digits.py

```diff
@@ -37,6 +70,11 @@ def digits(n, b=10):
         y.append(x)
         y.append(-b if n < 0 else b)
         y.reverse()
+        if bits is not None:
+            if len(y) - 1 < bits:
+                y = [b] + [0]*(bits - len(y) + 1) + y[1:]
```

Comment:
Wouldn't the whole point of the flag be to avoid computing unnecessary digits? I think this should be done in the above while loop.

Reply:
Oh nevermind, this is about giving more digits, not fewer.

Reply:
Sure sir. I will make the changes.

Reply:
Okay, sir. Going back to the previous code.